ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -975,6 +976,165 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+# TODO refactor some of these on_span_started/on_span_finished functions
+# that are used in other patched methods in this file into the utils module
+@with_traced_module
+def traced_chain_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+
+    def _on_span_started(span: Span):
+        inputs = get_argument_value(args, kwargs, 0, "input")
+        if integration.is_pc_sampled_span(span):
+            if not isinstance(inputs, list):
+                inputs = [inputs]
+            for idx, inp in enumerate(inputs):
+                if not isinstance(inp, dict):
+                    span.set_tag_str("langchain.request.inputs.%d" % idx, integration.trunc(str(inp)))
+                else:
+                    for k, v in inp.items():
+                        span.set_tag_str("langchain.request.inputs.%d.%s" % (idx, k), integration.trunc(str(v)))
+
+    def _on_span_finished(span: Span, streamed_chunks, error: Optional[bool] = None):
+        if not error and integration.is_pc_sampled_span(span):
+            if langchain_core and isinstance(instance.steps[-1], langchain_core.output_parsers.JsonOutputParser):
+                # it's possible that the chain has a json output parser
+                # this will have already concatenated the chunks into a json object
+
+                # it's also possible the json output parser isn't the last step,
+                # but one of the last steps, in which case we won't act on it here
+                # TODO (sam.brenner) make this more robust
+                content = json.dumps(streamed_chunks[-1])
+            else:
+                # best effort to join chunks together
+                content = "".join([str(chunk) for chunk in streamed_chunks])
```

Comment:
it's a bit tricky with chains, because there could be an output parser that _does_ turn the chunks into strings, or they could be straight up `BaseMessage` types (ie `AIMessage`). in our current LCEL `chain` patching, we just [stringify](https://github.com/DataDog/dd-trace-py/blob/baac738726a6b615cb3f79693b0b1f117b688f9d/ddtrace/contrib/internal/langchain/patch.py#L844) to avoid going case-by-case.